tools/azure-devtools/src/azure_devtools/perfstress_tests/_perf_stress_runner.py

```diff
@@ -226,3 +230,25 @@ def _print_status(self, title):
 
         self._operation_status_tracker = total_operations
         self.logger.info("{}\t\t{}\t\t{:.2f}".format(current_operations, total_operations, average_operations))
+
+    # Formats a number with a minimum number of significant digits.
+    # Digits to the left of the decimal point are always significant.
+    # Examples:
+    # - _format_number(0, 4) -> "0.000"
+    # - _format_number(12345, 4) -> "12,345"
+    # - _format_number(1.2345, 4) -> "1.235"
+    # - _format_number(0.00012345, 4) -> "0.0001235"
```

Comment:
Could we format this into a Python-style docstring - and maybe add a comment that this was explicitly ported across languages for consistency?
```python
def _format_number(self, value, min_significant_digits):
    """
    Formats a number with a minimum number of significant digits.

    This function has been ported from .NET for cross-language consistency.
    Digits to the left of the decimal point are always significant.
    Examples:
    - _format_number(0, 4) -> "0.000"
    - _format_number(12345, 4) -> "12,345"
    - _format_number(1.2345, 4) -> "1.235"
    - _format_number(0.00012345, 4) -> "0.0001235"
    """

```